cpp/benchmarks/string/convert_floats_benchmark.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../fixture/benchmark_fixture.hpp"
+#include "../synchronization/synchronization.hpp"
+
+#include <benchmark/benchmark.h>
+
+#include <cudf/column/column_view.hpp>
+#include <cudf/strings/convert/convert_floats.hpp>
+#include <cudf/types.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+
+#include <numeric>
+#include <random>
+#include <unordered_map>
+
+namespace {
+
+template <class FloatType>
+static const std::vector<FloatType>& get_float_numbers(int64_t array_size)
+{
+  static std::unordered_map<int64_t, std::vector<FloatType>> number_arrays;
+  auto& numbers = number_arrays[array_size];
+  if (numbers.size() == 0) {
+    numbers.reserve(array_size);
+    cudf::test::UniformRandomGenerator<FloatType> rand_gen(std::numeric_limits<FloatType>::min(),
+                                                           std::numeric_limits<FloatType>::max());
+    std::generate_n(
+      std::back_inserter(numbers), array_size, [&rand_gen]() { return rand_gen.generate(); });
+  }
+  return numbers;
+}
+
+template <class FloatType>
+static const std::vector<std::string>& get_floats_numbers_as_string(int64_t array_size)
+{
+  static std::unordered_map<int64_t, std::vector<std::string>> str_arrays;
+  auto& numbers_str = str_arrays[array_size];
+  if (numbers_str.size() == 0) {
```

Comment:
Is this map necessary? Are the vectors re-used in the gbenchmarks?

Reply:
Yes, the numbers/strings vectors are reused for the case of float32/float64 conversion benchmarks.

Reply:
Maybe not in this benchmark. I observed that for each size the number/string vector is used only once.

Reply:
I see the case for the number vector since it is used for both the to/from benchmarks. But the string one is used only in the from-string benchmark. So I think this one can be simplified.